code/modules/fishing/fishing_portal_machine.dm

```diff
@@ -11,6 +11,46 @@
 
 	///The current fishing spot loaded in
 	var/datum/component/fishing_spot/active
+	///A list of fishing spot it's linked to with a multitool.
+	var/list/linked_fishing_spots
+	///The maximum number of fishing spots it can be linked to
+	var/max_fishing_spots = 1
+	///If true, the fishing portal can stay connected to a linked fishing spot even on different z-levels
+	var/long_range_link = FALSE
+
+/obj/machinery/fishing_portal_generator/Initialize(mapload)
+	. = ..()
+	var/static/list/tool_screentips = list(
+		TOOL_MULTITOOL = list(
+			SCREENTIP_CONTEXT_LMB = "Link",
+			SCREENTIP_CONTEXT_RMB = "Unlink fishing spots"
+		),
+	)
+	AddElement(/datum/element/contextual_screentip_tools, tool_screentips)
+	ADD_TRAIT(src, TRAIT_UNLINKABLE_FISHING_SPOT, INNATE_TRAIT)
+
+/obj/machinery/fishing_portal_generator/Destroy()
+	deactivate()
+	linked_fishing_spots = null
+	return ..()
+
+///Higher tier parts let you link to more fishing spots at once and eventually let you connect through different zlevels.
+/obj/machinery/fishing_portal_generator/RefreshParts()
+	. = ..()
+	max_fishing_spots = 0
+	long_range_link = FALSE
+	for(var/datum/stock_part/matter_bin/matter_bin in component_parts)
+		max_fishing_spots += matter_bin.tier * 0.5
+	max_fishing_spots = round(max_fishing_spots)
```

Comment:
Did you intend to make it so tier 1 has 0 fishing spots? `round(x)` is the same as `floor(x)`. You probably want the following here
```suggestion
	max_fishing_spots = ROUND_UP(max_fishing_spots)
```